spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -93,7 +94,8 @@
 
 	private final RedisTemplate<String, RedisLock> redisTemplate;
 
-	private final ThreadLocal<List<RedisLock>> threadLocks = new ThreadLocal<List<RedisLock>>();
+	private final ThreadLocal<Set<RedisLock>> weakThreadLocks = new ThreadLocal<Set<RedisLock>>();
+	private final ThreadLocal<List<RedisLock>> hardThreadLocks = new ThreadLocal<List<RedisLock>>();
```

Comment:
Well yes - you are right about second concern - without variable that `equals` will not take place or makes no sense, so doc update should be enough. Immutable properties of the lock - if extract locked time and thread name from hash then we need do the same for `equals` to do not break contract. So then we won't be able to relay on hashCode and equality and always would need to compare two additional fields (inside and outside of this Registry)...